my-sms/spec/factories/courses.rb

```diff
@@ -12,12 +14,12 @@
     end_date 2.years.ago.to_date
     institute
 
-    factory :course_demo, class: :course do
+    trait :demo do
       name { Faker::Educator.course_name }
       description { Faker::Lorem.paragraph }
       start_date Faker::Date.between(from: '2014-01-1', to: '2014-06-15')
       end_date Faker::Date.between(from: '2014-06-16', to: '2014-12-30')
-      institute { association :institute_demo }
+      institute { with_institute ? association(:institute_demo) : nil }
```

Comment:
Can also just override the `institute` field when creating one, or create a new trait.

eg.

```ruby
create(:course, :demo, institute: nil)
```

OR

```ruby
trait :no_institute do
  institute { nil }
end

...


create(:course, :demo, :no_institute)
```

Reply:
Thats way better thanks. Went for option 2.
Fixed in f5201bd4bc1bc0d6884387f28c006def12d6625e